includes/class-freemius.php

```diff
@@ -918,9 +958,49 @@ private function _find_caller_plugin_file( $is_init = false ) {
 				'path' => $id_slug_type_path_map[ $this->_module_id ]['path'],
 			);
 
-			return $id_slug_type_path_map[ $this->_module_id ]['path'];
+			return $this->get_absolute_path( $id_slug_type_path_map[ $this->_module_id ]['path'] );
 		}
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 1.2.2.11
+         *
+         * @param string $path
+         *
+         * @return string
+         */
+		private function get_relative_path( $path ) {
+            return $this->_plugin_basename .
+                ( $this->is_theme() ?
+                    '/' . basename( $path ) :
+                    '' );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 1.2.2.11
+         *
+         * @param string      $path
+         * @param string|bool $module_type
+         *
+         * @return string
+         */
+		private function get_absolute_path( $path, $module_type = false ) {
+		    $is_plugin = empty( $module_type ) ?
+                $this->is_plugin() :
+                ( WP_FS__MODULE_TYPE_PLUGIN === $module_type );
+
+		    $module_root_dir = fs_normalize_path( $is_plugin ?
+                WP_PLUGIN_DIR :
+                get_theme_root() );
+
+		    if ( 0 !== strpos( $path, $module_root_dir ) ) {
+		        $path = ( $module_root_dir . '/' . $path );

```

Comment:
@fajardoleo I'd recommend using `$path = trailingslashit( $module_root_dir ) . $path;` instead to avoid situations where some environments may already have a trailing slash at the end of the plugin/theme path variable.